lib/rubygems/commands/install_command.rb

```diff
@@ -260,7 +261,8 @@ def show_install_errors(errors) # :nodoc:
     errors.each do |x|
       return unless Gem::SourceFetchProblem === x
 
-      msg = "Unable to pull data from '#{x.source.uri}': #{x.error.message}"
```

Comment:
The `x.source.uri` could be a URI with a credential so `Gem::PrintableUri` is used here to make sure that sensitive information will not be exposed.